### DISCUSSION TITLE
Korean Links

### DISCUSSION THREAD

**Discussion by Alstruit** [iptv-org/community#13] [Show and tell]
Found some neat apis or websites that can be "scrapped" for m3u8s.

Naver TV:
API: https://api.tv.naver.com/api/open/live/v2/player/playback?liveId=xxx&countryCode=yy
Requirements:
Needs a valid ID and 2 digit Country Code

ID: 3077128 ->  Screen Golfzon
ID: 3284836 -> Tomato News
ID: 3214537 -> Yonhap News

YTN:
YTN 24/7 -> https://www.ytn.co.kr/_hd/hd_live.html (Unlisted url Youtube link expires after 1 day)
YTN 24/7 -> https://www.youtube.com/channel/UChlgI3UHCOnwUGzWzbJ3H5w/live (Geo-locked only against US)
YTN2 -> https://ytn2.ytn.co.kr/hd/hd_onair.php -> https://ytn2.ytn.co.kr/hd/cdnurl.js?v=2031 (json) -> hls

TV Chosun 1 & 2 (Stream force refreshes once in while)
http://broadcast.tvchosun.com/onair/on.cstv
http://broadcast.tvchosun.com/onair/on2.cstv
TV chosun 1
http://onair.cdn.tvchosun.com/origin1/_definst_/tvchosun_s1/playlist.m3u8
TV chosun 2
http://onair2.cdn.tvchosun.com/origin2/_definst_/tvchosun_s3/playlist.m3u8

KBS News D:
https://news.kbs.co.kr/api/getNewsLiveStreamUrl?channel=news24 (json) -> streamUrl (Expires in 2 days).

KBS1
https://vod.kbs.co.kr/embed_live.html?ch_code=11 -> m3u8 in 'https://1tvhd.gscdn... (var channel -> service_url)
KBS2
https://vod.kbs.co.kr/embed_live.html?ch_code=12 -> m3u8 in https://2tvhd.gscdn... (var channel -> service_url)
KBS World
https://vod.kbs.co.kr/embed_live.html?ch_code=14 -> m3u8 in https://world.gscdn... (var channel -> service_url)
KBS Drama
https://vod.kbs.co.kr/embed_live.html?ch_code=N91 -> m3u8 in https://kbsndrama.gscdn... (var channel -> service_url)
KBS Joy
https://vod.kbs.co.kr/embed_live.html?ch_code=N92 -> m3u8 in https://kbsnjoy.gscdn... (var channel -> service_url)
KBS Life
https://vod.kbs.co.kr/embed_live.html?ch_code=N93 -> m3u8 in https://kbsnlife.gscdn... (var channel -> service_url)
KBS Story
https://vod.kbs.co.kr/embed_live.html?ch_code=N94 -> m3u8 in https://kbsnw.gscdn... (var channel -> service_url)
KBS Kids
https://vod.kbs.co.kr/embed_live.html?ch_code=N96 -> m3u8 in https://kbsnkids.gscdn... (var channel -> service_url)

**Comment by AzeemHadzrie** [comment]
https://tvlive.sbs.co.kr/sbsch6/sbsch61.stream/playlist.m3u8?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJleHAiOjE2NjQzMjk5MzEsInBhdGgiOiIvc2JzY2g2MS5zdHJlYW0iLCJkdXJhdGlvbiI6MTgwLCJ1bm8iOiJiY2ZhMjdkMi05N2RjLTQ5MTYtOGUyNS1mZDU1ZWEyYzNhNGEiLCJpYXQiOjE2NjQzMjk3NTF9.xTP6Cixt4-m0npew_474rhPLTo7Gu2lFctO7G3xAXMc Requires Korean VPN.

**Comment by BellezaEmporium** [comment]
> 

Not unlisted : https://www.youtube.com/c/ytnnews24/live

**Reply by Alstruit** [reply]
It was unlisted before last time I checked, but it is still geo-locked against US.

**Reply by BellezaEmporium** [reply]
Indeed, just tested with a US IP and it's indeed locked.

**Comment by BellezaEmporium** [comment]
> Found some neat apis or websites that can be "scrapped" for m3u8s.
> 
> Naver TV: API: https://api.tv.naver.com/api/open/live/v2/player/playback?liveId=xxx&countryCode=yy Requirements: Needs a valid ID and 2 digit Country Code
> 
> ID: 3077128 -> Screen Golfzon ID: 3284836 -> Tomato News ID: 3214537 -> Yonhap News

Gonna check the Naver one.

**Reply by BellezaEmporium** [reply]
https://nv-streams.onrender.com/scraper?channel=3077128 ;)

**Comment by sooyaaa131995** [comment]
@LaneSh4d0w Sorry to trouble you, but how you make scraper of seezn?

**Reply by BellezaEmporium** [reply]
I use SeeZN's API. I provide the ID along with the usual parameters they ask for, and I grab the HLS URL.

**Reply by sooyaaa131995** [reply]
what is their API url and the parameter and ids used? plan tp create my own version of it.

**Reply by BellezaEmporium** [reply]
```
# get actual time
today = datetime.now()
# format time to fit seezn standards by trimming the microseconds (not a smiley uwu)
timestamp_seezn = today.strftime("%Y%m%d%H%M%S%f")[:-3]
transactionid_seezn = timestamp_seezn + "000000000000001"

@app.route("/scraper")
@limiter.limit("1/second")
def scrape():
    if not request.args.get("channel"):
        return "You didn't provide a channel number"
    
    r = requests.get("https://api.seezntv.com/svc/menu/app6/api/epg_preplay?ch_no=" + request.args.get("channel") + "&bit_rate=S&bit_rate_option=4000&is_5g=Y&hevc_yn=N&ch_5g_code=0&protocol=https&istest=1",
    headers={'timestamp': timestamp_seezn, 
             'transactionId': transactionid_seezn,
             'X-APP-VERSION': "104.0",
             'X-DEVICE-MODEL': "Firefox",
             'X-DEVICE-TYPE': 'PCWEB',
             'X-OS-TYPE': "Windows",
             'X-OS-VERSION': "NT 10.0",
             'HTTP_CLIENT_IP': "undefined"})
    if not r.json()['data'].get('live_url'):
        return "The channel number you've provided doesn't match Seezn's data."
    return redirect(r.json()['data'].get('live_url'))
 ```

**Reply by sooyaaa131995** [reply]
thank very much :)

**Reply by sooyaaa131995** [reply]
already made PHP version, working so good:
`<?php
$c = isset($_GET["c"]) ? $_GET["c"] : 280;
date_default_timezone_set("Asia/Seoul");
$d = new DateTime();
$dt = $d->modify('-3 microseconds');
$timestamp_seezn = $dt->format('YmdHisv');
$opts = array(
  'http'=>array(
    'method'=>"GET",
    'header'=>"Accept-language: en\r\n" .
              "timestamp': $timestamp_seezn\r\n" . 
              "transactionId: '. $timestamp_seezn .'000000000000001\r\n" . 
              "X-APP-VERSION: 104.0\r\n" . 
              "X-DEVICE-MODEL: Firefox=bar\r\n" . 
              "X-DEVICE-TYPE: PCWEB\r\n" . 
              "X-OS-TYPE: Windows=bar\r\n" . 
              "X-OS-VERSION: NT 10.0\r\n" . 
              "HTTP_CLIENT_IP: 106.255.213.187=bar\r\n" . 
              "Cookie: foo=bar\r\n" . 
             "Referer: https://www.seezntv.com/liveTv?ch_no='. $c .'&category_id=1&channel_type=epg\r\n"
  )
);

$context = stream_context_create($opts);

$seezn = file_get_contents('https://api.seezntv.com/svc/menu/app6/api/epg_preplay?ch_no='. $c .'&bit_rate=S&bit_rate_option=1000&is_5g=Y&hevc_yn=N&ch_5g_code=0&protocol=https&istest=1', false, $context);
preg_match('/live_url":"(.*?)"/s', $seezn, $matches);
$l = $matches[1];
header("HTTP/1.1 301 Moved Permanently");
header("Location: $l");
exit();
?>`

**Comment by Alstruit** [comment]
Here's one for MBN.kr
https://www.mbn.co.kr/vod/onair:
(1080p)
https://www.mbn.co.kr/player/mbnStreamAuth_new_live.mbn?vod_url=https://hls-live.mbn.co.kr/mbn-on-air/1000k/playlist.m3u8 ->  returns generated m3u8.
(480p)
https://www.mbn.co.kr/player/mbnStreamAuth_new_live.mbn?vod_url=https://hls-live.mbn.co.kr/mbn-on-air/600k/playlist.m3u8 8 ->  returns generated m3u8.

**Comment by BellezaEmporium** [comment]
TV Chosun links only need a referrer to work : http://broadcast.tvchosun.com/onair/on.cstv

**Comment by Alstruit** [comment]
This [website](https://blog.bsod.kr/137) provide further information about the reginal channels for KBS.

https://cfpwwwapi.kbs.co.kr/api/v1/landing/live/channel_code/[Region code]_[Name of Channel]
or
https://cfpwwwapi.kbs.co.kr/api/v1/landing/live/channel_code/[Channel Code]

Channel Code:
11 -> KBS1
12 -> KBS2
Region code (KBS 1 only):
10 -> Busan
20 -> Changwon
21 -> Jinju
30 -> Daegu
31 -> Andong
32 -> Pohang
40 -> Gwangju
41 -> Mokpo
43-> Suncheon
50-> Jeonju
60 -> Daejeon
70 -> Chungju
80 -> Chuncheon
81 -> Gangneung
82 -> Wonju
90 -> Jeju
(Update 10.20.23)
11 -> Ulsan

**Reply by BellezaEmporium** [reply]
Works quite fine actually. If you just put an intermediary in-between, it works perfectly. Though it would need to have a Korean IP in order to access the stream.
